lib/Bindings/Python/dialects/om.py

```diff
@@ -0,0 +1,103 @@
+#  Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+#  See https://llvm.org/LICENSE.txt for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from __future__ import annotations
+
+from ._om_ops_gen import *
+from .._mlir_libs._circt._om import Evaluator as BaseEvaluator, Object
+
+from circt.ir import Attribute, Diagnostic, DiagnosticSeverity, Module, StringAttr
+from circt.support import attribute_to_var, var_to_attribute
+
+import sys
+import logging
+from dataclasses import fields
+from typing import TYPE_CHECKING, Any, TypeVar
+
+if TYPE_CHECKING:
+  from _typeshed.stdlib.dataclass import DataclassInstance
+
+
+# Define the Evaluator class by inheriting from the base implementation in C++.
+class Evaluator(BaseEvaluator):
+
+  def __init__(self, mod: Module) -> None:
+    """Instantiate an Evaluator with a Module."""
+
+    # Call the base constructor.
+    super().__init__(mod)
+
+    # Set up logging for diagnostics.
+    logging.basicConfig(
+        format="[%(asctime)s] %(name)s (%(levelname)s) %(message)s",
+        datefmt="%Y-%m-%d %H:%M:%S",
+        level=logging.INFO,
+        stream=sys.stdout,
+    )
+    self._logger = logging.getLogger("Evaluator")
+
+    # Attach our Diagnostic handler.
+    mod.context.attach_diagnostic_handler(self._handle_diagnostic)
+
+  def instantiate(self, cls: type["DataclassInstance"],
+                  *args: Any) -> "DataclassInstance":
+    """Instantiate an Object with a dataclass type and actual parameters."""
+
+    # Convert the class name and actual parameters to Attributes within the
+    # Evaluator's context.
+    with self.module.context:
+      # Get the class name from the provided dataclass name.
+      class_name = StringAttr.get(cls.__name__)
+
+      # Get the actual parameter Attributes from the supplied variadic
+      # arguments. This relies on the circt.support helpers to convert from
+      # Python objects to Attributes.
+      actual_params = var_to_attribute(list(args))
```

Comment:
Nice! :)

Reply:
Yeah this was what confused me initially, `args` is actually a tuple not a list, but this is what I wanted--pass it directly to `var_to_attribute`.